### ISSUE TITLE
Thread-safety issue when encoding blurhash in multiple threads

### ISSUE THREAD

**Issue by nattster** [woltapp/blurhash-python#18]
When trying to encode multiple images in parallel in thread-pool executor,
`blurhash.encode` returns blurhash value from other thread.

## Versions
* Python 3.11.3
* blurhash-python==1.2.0

## Code to reproduce

```python
# reproduce.py
import blurhash
from PIL import Image
import glob
from concurrent.futures import ThreadPoolExecutor, as_completed

for i in range(10):
    print(i)
    with ThreadPoolExecutor(max_workers=5) as executor:

        def func(filename):
            im = Image.open(filename)
            return blurhash.encode(im, x_components=4, y_components=3)

        tasks = [executor.submit(func, filename) for filename in glob.glob("*.jpg")]
        unique_hashes = set()
        for h in as_completed(tasks):
            hash_value = h.result()
            if hash_value in unique_hashes:
                print("duplicate hash found " + hash_value)
            unique_hashes.add(hash_value)
```

To reproduce:
* Save above script in a directory with following 5 images:
    * https://blurha.sh/e08adedc9f82ade9f9d5.jpg
    * https://blurha.sh/12c2aca29ea896a628be.jpg
    * https://blurha.sh/a355ab362a07a267e078.jpg
    * https://blurha.sh/ea9e499f8080ce9956a8.jpg
    * https://blurha.sh/2af1b155f29d7089e3d9.jpg
* Run `python3 reproduce.py`
* The script will calculate blurhash in parallel with up to 5 threads. It stores blurhash result in `set()` and reports duplicate if it finds any.

```
python3 reproduce.py
0
1
duplicate hash found LPPGdFbI?ws.-;ofM|R+OFV@r;xu
2
3
duplicate hash found LKO2?V%2Tw=^]~RBVZRi};RPxuwH
4
duplicate hash found LKO2?V%2Tw=^]~RBVZRi};RPxuwH
5
6
7
8
9
```

**Comment by lautat** [comment]
Thank you for the detailed bug report! I was able to reproduce the issue, although not very reliably. I might have a clue what's causing the issue, and I'll try to fix it this week.

**Comment by lautat** [comment]
Thanks once again for the report! I fixed the issue in #19, and published a new version to PyPI: https://pypi.org/project/blurhash-python/1.2.1/
